cloud_info_provider/providers/static.py

```diff
@@ -214,17 +217,49 @@ def _get_defaults(self, what, which, prefix=''):
 
         return copy.deepcopy(defaults)
 
+    @staticmethod
+    def _set_defaults(values, defaults, prefix):
+        for k, v in defaults.items():
+            prefixed_key = ''.join((prefix, k))
+            if not values.get(prefixed_key):
+                values[prefixed_key] = v
+        return values
+
     def get_image_defaults(self, prefix=False):
         prefix = 'image_' if prefix else ''
         return self._get_defaults('compute', 'images', prefix=prefix)
 
     def get_template_defaults(self, prefix=False):
         prefix = 'template_' if prefix else ''
-        return self._get_defaults('compute', 'templates', prefix=prefix)
+        defaults = {
+            'network_out': True,
+        }
+        return self._set_defaults(self._get_defaults('compute', 'templates',
+                                                     prefix=prefix),
+                                  defaults, prefix)
 
     def get_compute_endpoint_defaults(self, prefix=False):
         prefix = 'compute_' if prefix else ''
-        return self._get_defaults('compute', 'endpoints', prefix=prefix)
+        defaults = {
+            'service_production_level': 'production',
+            'service_capabilities': ['executionmanagement.dynamicvmdeploy',
+                                     'security.accounting'],
+            'capabilities': ['executionmanagement.dynamicvmdeploy',
+                             'security.accounting'],
+            'production_level': 'production',
+            'api_endpoint_technology': 'webservice',
+            'api_authn_method': 'oidc',
+            'total_ram': 0,
+            'total_cores': 0,
+            'max_dedicated_ram': 0,
+            'min_dedicated_ram': 0,
+            'failover': False,
+            'live_migration': False,
+            'vm_backup_restore': False,
+        }
+        return self._set_defaults(self._get_defaults('compute', 'endpoints',
```

Comment:
`_set_defaults` from `_get_defaults` using `defaults`, it's quite confusing, could it be possible to comment so that what is happening is more easily understandable without scratching its head?

Reply:
Re-named all the methods to bring some clarity, hope is better now.

Reply:
yes, thanks!